diff -ur msgpack-0.5.5.orig/config.m4 msgpack-0.5.5/config.m4
--- msgpack-0.5.5.orig/config.m4
+++ msgpack-0.5.5/config.m4
@@ -4,17 +4,6 @@
 dnl Remove where necessary. This file will not work
 dnl without editing.
 
-dnl Check PHP version:
-
-AC_MSG_CHECKING(PHP version)
-AC_TRY_COMPILE([#include "$phpincludedir/main/php_version.h"], [
-#if PHP_MAJOR_VERSION < 5
-#error  this extension requires at least PHP version 5 or newer
-#endif
-],
-[AC_MSG_RESULT(ok)],
-[AC_MSG_ERROR([need at least PHP 5 or newer])])
-
 dnl If your extension references something external, use with:
 
 PHP_ARG_WITH(msgpack, for msgpack support,
diff -ur msgpack-0.5.5.orig/msgpack_convert.c msgpack-0.5.5/msgpack_convert.c
--- msgpack-0.5.5.orig/msgpack_convert.c
+++ msgpack-0.5.5/msgpack_convert.c
@@ -34,7 +34,7 @@
         return SUCCESS;                                                   \
     }
 
-inline int msgpack_convert_long_to_properties(
+static zend_always_inline int msgpack_convert_long_to_properties(
     HashTable *ht, HashTable **properties, HashPosition *prop_pos,
     uint key_index, zval *val, HashTable *var)
 {
@@ -133,7 +133,7 @@
     return zend_hash_index_update(ht, key_index, &val, sizeof(val), NULL);
 }
 
-inline int msgpack_convert_string_to_properties(
+static zend_always_inline int msgpack_convert_string_to_properties(
     zval *object, char *key, uint key_len, zval *val, HashTable *var)
 {
     zval **data = NULL;
