--- groonga-3.0.4/lib/CMakeLists.txt.orig
+++ groonga-3.0.4/lib/CMakeLists.txt
@@ -34,7 +34,11 @@
   COMPILE_FLAGS "${GRN_CXX_COMPILE_FLAGS}")
 
 add_library(libgroonga SHARED ${LIBGROONGA_SOURCES} ${LIBGRNDAT_SOURCES})
-set_target_properties(libgroonga PROPERTIES OUTPUT_NAME "groonga")
+set_target_properties(libgroonga
+  PROPERTIES
+  OUTPUT_NAME "groonga"
+  INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${LIB_DIR}"
+)
 if(GRN_WITH_STATIC)
   add_library(libgroonga_static STATIC ${LIBGROONGA_SOURCES} ${LIBGRNDAT_SOURCES})
   set_target_properties(libgroonga_static PROPERTIES OUTPUT_NAME "groonga")
